### DISCUSSION TITLE
Example project using Huggingface task model (e.g. for POS tagging)

### DISCUSSION THREAD

**Discussion by arsalan993** [explosion/spaCy#9611] [New Features & Project Ideas]
I am trying to use the following POS tagger for the Spanish language from hugging face` mrm8488/bert-spanish-cased-finetuned-pos-16-tags` but I can't find a proper method for adding hugging face model to the NLP pipeline. 

Furthermore i want to use these POS tags in the textrank which I will eventually also add the pipeline

Previously i was stanza which was quite straight forward 
```
nlp = spacy_stanza.load_pipeline("es")
nlp.add_pipe('textrank', last=True)
nlp(sentences)
```
for hugging face i cant even find a straight forward documentation.

Mean while you guys have mentioned this **This pipeline component lets you use transformer models in your pipeline. It supports all models that are available via the HuggingFace transformers library.**  over here [spacy doc](https://spacy.io/api/transformer)

**Comment by honnibal** [comment]
It's true that there's a gap here, both in the process and the documentation.

It comes from the ambiguity about what a "model" is. When we wrote, "supports all HF models", we're talking about using transformers to power the token-to-vector encoding. This is what reuseable transformers such as bert-base-uncased etc do.

What you're looking for is a specific task (I.e. part-of-speech tagging), using the trained POS tagger from their hub. This is something we don't have an example for yet.

For POS tagging I think we can come up with a pretty straight-forward solution. But the same won't necessarily be true of every task model on the hub.

**Comment by arsalan993** [comment]
If you guys can share POS tagger based solution using the model i have shared above. Which i can further add with textrank in nlp pipeline then plz do share..
The reason i am not using your pretrained spanish tranaformer model is because it is trained on some news dataset which is returning  inaccurate labels for POS. While these hugging face models give extremely  accurate  results considering the face that they are trained on some other dataset.

It will be extremely  kind of you guys if you can help me in this.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by honnibal** [comment]
We'll add "Example using Huggingface fine-tuned model to power component" to our list of examples we'd like to produce, but the purpose of the example will be to help the community more generally. I can't promise how well it'll address your specific use-case. I doubt we'd address the textrank requirement, for instance.

**Comment by arsalan993** [comment]
Looking forward.  If that could be generalize in a way that i can add text rank to nlp pipeline then that will be awsome.

Please share link of that example here if possible.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by polm** [reply]
Maybe you already saw it, but [PyTextRank](https://github.com/DerwenAI/pytextrank) has a spaCy plugin. Does it not do what you need?

**Comment by honnibal** [comment]
Moving this to the "New features and project ideas" section of the discussion board.
